### PR TITLE
Add Workflow Dispatch Event by ID and FileName

### LINKED ISSUE #1639: Can't create workflow dispatch event by filename
Noticed that, [GitHub API docs: Create a workflow dispatch event](https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#create-a-workflow-dispatch-event), let's you create dispatch event by both workflowID and Filename.

However, [PR #1623 ](https://github.com/google/go-github/pull/1623) adds support for just workflowID, with **CreateWorkflowDispatchEvent**.

Can we split it into two _funcs_ for each type, as that's the general implementation in other APIs?

### PR DESCRIPTION
**Pull request by n1lesh**
Fixes #1639

### CODE REVIEW

File: github/actions_workflows.go
Comment (gmlewis) [inline github/actions_workflows.go:154]: @n1lesh - can you please explain to me what this extra `free-pro-team@latest` is and why you are adding these to the URL?

I have a utility in this repo that updates URLs, and it's not going to find these.

Thank you!

File: github/actions_workflows.go
Comment (gmlewis) [inline github/actions_workflows.go:155]: We typically pass mandatory fields by value in this repo... so I'm thinking that these two exported functions should be left as they were and not use pointers.

However, I'm totally fine to let your `create...` helper function use a pointer since that stays internal to the package.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @n1lesh !

One question and one comment, please.

File: github/actions_workflows.go
Comment (n1lesh) [reply github/actions_workflows.go:155]: I actually took a little hint from `CreateUserImpersonation` from admin_users.go that expects pointer to `ImpersonateUserOptions`. I will update the two funcs to expect value though.

File: github/actions_workflows.go
Comment (n1lesh) [reply github/actions_workflows.go:154]: For me, the docs open with `free-pro-team@latest` in the url, even when I open the old url. Maybe they changed it, or it's just for me?

Comment (gmlewis) [reply github/actions_workflows.go:154]: Ah!  It does for me now too... it looks like I'll have to update all the URLs again at some point.  :joy:
Thanks, @n1lesh !

**Review by gmlewis — APPROVED** [review]
Thank you, @n1lesh !
LGTM.

Awaiting second LGTM before merging.

**Comment by gmlewis** [comment]
@erikkn - it would be nice to get all breaking changes into the next major release (v33.0.0) so that we don't have to bump the major version more often than necessary. If you feel like reviewing this PR, that would be greatly appreciated.

**Comment by gmlewis** [comment]
friendly ping @wesleimp

**Comment by gmlewis** [comment]
@nightlark - if you are available for a review, it would be greatly appreciated.

**Review by erikkn — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @erikkn !
Merging.

**Pull request merged**
